### ISSUE TITLE
Enable possibility to use a static file as logo

### ISSUE THREAD

**Issue by BMourguesFieldbox** [fabiocaccamo/django-admin-interface#355]
In the current version, the logo can be user uploaded. 
However, this means that the app has to handle user uploaded documents and serve them. 

When building the admin for business client, the logo will most likely be in static files, and I think it could save some complexity to have the possibility to use a CharField that would point to a static file `app/logo.png` for example. 

Then the template would be with a condition 

```jinja2

{% if theme.use_static_logo %}}
    ...
    {% static theme.static_logo_path %}}
 {% else %}
     # The current behaviour
 {% endif %}
 ```
 
 Wdyt ?

**Comment by fabiocaccamo** [comment]
@BMourguesFieldbox thank you for this suggestion, it's indeed interesting, I will think more about it.

**Comment by FraCata00** [comment]
great solution, instead of `CharField`, i'll be use a `FilePathField`

from django docs [FilePathField](https://docs.djangoproject.com/en/4.2/ref/models/fields/#filepathfield)
![image](https://github.com/fabiocaccamo/django-admin-interface/assets/78490028/5d384593-6619-4a92-a6a5-5f7b263b1ee4)

**Comment by BMourguesFieldbox** [comment]
> great solution, instead of `CharField`, i'll be use a `FilePathField`
> 
> from django docs [FilePathField](https://docs.djangoproject.com/en/4.2/ref/models/fields/#filepathfield) ![image](https://private-user-images.githubusercontent.com/78490028/313637854-5d384593-6619-4a92-a6a5-5f7b263b1ee4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3NTkxODIsIm5iZiI6MTcxMDc1ODg4MiwicGF0aCI6Ii83ODQ5MDAyOC8zMTM2Mzc4NTQtNWQzODQ1OTMtNjYxOS00YTkyLWE2YTUtNWY3YjI2M2IxZWU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE4VDEwNDgwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhNWJiNzU2NTRmYzJmN2JkM2RhMWE4ZmNhMTFkMTA4ZDg5ZjRiYWRlNGFlNWE2NTlmNmE2ZWZkNzEwM2MwNTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eYwy6n7c0KIwR2hbN0qktkQzm4ggcEgLdazGgbzP0lA)

Good catch ! Didn't think of it, but sure thanks for the suggestion!

**Comment by FraCata00** [comment]
> > great solution, instead of `CharField`, i'll be use a `FilePathField`
> > from django docs [FilePathField](https://docs.djangoproject.com/en/4.2/ref/models/fields/#filepathfield) ![image](https://private-user-images.githubusercontent.com/78490028/313637854-5d384593-6619-4a92-a6a5-5f7b263b1ee4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3NTkxODIsIm5iZiI6MTcxMDc1ODg4MiwicGF0aCI6Ii83ODQ5MDAyOC8zMTM2Mzc4NTQtNWQzODQ1OTMtNjYxOS00YTkyLWE2YTUtNWY3YjI2M2IxZWU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE4VDEwNDgwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhNWJiNzU2NTRmYzJmN2JkM2RhMWE4ZmNhMTFkMTA4ZDg5ZjRiYWRlNGFlNWE2NTlmNmE2ZWZkNzEwM2MwNTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eYwy6n7c0KIwR2hbN0qktkQzm4ggcEgLdazGgbzP0lA)
> 
> Good catch ! Didn't think of it, but sure thanks for the suggestion!

It's better, because you have more control instead a classic `CharField`

**Comment by fabiocaccamo** [comment]
@FraCata00 @BMourguesFieldbox to allow users to decide which directory to choose the file from, a conf setting must be added for setting the field path, but this would impact migrations.

**Comment by FraCata00** [comment]
@fabiocaccamo the users must do a 'migrate' command

**Comment by fabiocaccamo** [comment]
@FraCata00 indeed, but that migration will be added to `admin_interface` app, and when in the future this app will have new migrations there will be some conflicts.

**Comment by FraCata00** [comment]
@fabiocaccamo why I'll be in a conflict migrations?

**Comment by fabiocaccamo** [comment]
@FraCata00 I try to explain you step by step:

- now this package has 30 migrations
- if we add this new field it will have 31 migrations
- you install the package with 31 migrations, you customize the path setting, then you will have to run `makemigrations`, and as result you will have 32 migrations inside `admin_interface` package, but the "real" application that you install from pip has only 31 migrations
- a new feature that relies on migrations gets developed, the app migrations increase to 32, you upgrade the package and migration conflict error! because you already have migration 32.

**Comment by FraCata00** [comment]
@fabiocaccamo but if you customize the path setting, should not run `makemigrations`, is a features of FilePathField. We can add a callable to dynamically return the path for the new field, any opinions?

**Comment by fabiocaccamo** [comment]
@FraCata00 should test it very well:

- [ ] add the field
- [ ] set path to a callable that return a value taken from conf settings
- [ ] run make migrations and migrate
- [ ] change the setting value
- [ ] run make migrations and ensure that no new migrations have been created

**Comment by FraCata00** [comment]
@fabiocaccamo yessir, after work i'll try it okay?

**Comment by FraCata00** [comment]
@fabiocaccamo I confirm that no new migrations are crreated if the value in settings changed 👍🏻

**Comment by fabiocaccamo** [comment]
@FraCata00 good news!

**Comment by FraCata00** [comment]
@fabiocaccamo @BMourguesFieldbox I tested peronally yesterday evening in my branch to introduce this functionality
Just a little bit of testing it better, and i'll do a PR 👍🏻

**Comment by fabiocaccamo** [comment]
@FraCata00 I would add only a new `logo_static` field, then if both are defined in the backend, the media will have the priority.

**Comment by FraCata00** [comment]
@fabiocaccamo oh, maybe is a better solution

**Comment by FraCata00** [comment]
@fabiocaccamo sorry for the time *I'm in the office tomorrow morning 😆 ), check PR to solve this issue #378
